---
title: "Assignment 1"
output:
  word_document: default
  html_document: default
date: "2022-10-01"
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE, echo=FALSE}
options(repos = list(CRAN="http://cran.rstudio.com/"))
require("knitr")
opts_knit$set(root.dir <- "~/Desktop/uf/FALL22/bioinformatics")
library(org.Hs.eg.db) 
library(magrittr)


install.packages("htmltools")
library(htmltools)
BiocManager::install("DESeq2")
BiocManager::install("apeglm")
BiocManager::install("EnhancedVolcano")
library("DESeq2")
library(ggplot2)
install.packages("dplyr")
library(dplyr)
install.packages("tidyverse")

data_dir <- file.path(root.dir, "data")
data_file <- file.path(data_dir,"GSE211669_20_04_2020.HTSeq.all.raw.txt")


expression_df <- readr::read_tsv(data_file) %>%
  tibble::column_to_rownames("GeneID")

library(stringr)
df <- str_split_fixed((rownames(expression_df)),"\\|",n=9)
rownames(expression_df) <- make.names(df[,8], unique = "TRUE")

length(unique(df[,8]))

mapped_list <- mapIds(
  org.Hs.eg.db, # Replace with annotation package for your organism
  keys = df[,1],
  keytype = "ENSEMBL", # Replace with the type of gene identifiers in your data
  column = "SYMBOL", # The type of gene identifiers you would like to map to
)

mapped_df <- mapped_list %>%
  tibble::enframe(name = "Ensembl", value = "Symbol")
  # enframe() makes a `list` column; we will simplify it with unnest()
  # This will result in one row of our data frame per list item

sum(is.na(mapped_df$Symbol))


```

## 1

```{r}
dim(expression_df) # matrix size
nrow(expression_df) # number of genes

logExpression_df <- log(expression_df + 1, 2) #log scaling - trying to avoid dealing w/ 0

perGeneExpressionRanges <- apply(logExpression_df,1, max) - apply(logExpression_df,1,min) #calculate per-gene expression ranges


plot(density(perGeneExpressionRanges))#density plot

```

The expression matrix has a size of 57773 x 131, with 131 observed
patients and 57773 observed genes. The most common gene expression range
is e\^3 â‰ˆ 20, suggesting a majority of the genes expressed were observed
in similar quantities amongst test subjects.

## 2

```{r}
metadata56 <- read.table("data/GSE211669-GPL24676_series_matrix.txt" , fill = TRUE, skip = 27, sep = "\t")
rowNames <- make.names(metadata56[,1],unique=TRUE)
rownames(metadata56) <- rowNames
metadata56<-metadata56[,-c(1)]
metadata56<-metadata56[-c(1:4),]

metadata75 <- read.table("data/GSE211669-GPL11154_series_matrix.txt" , fill = TRUE, skip = 27, sep = "\t")
rowNames <- make.names(metadata75[,1],unique=TRUE)
rownames(metadata75) <- rowNames
metadata75<-metadata75[,-c(1)]
metadata75<-metadata75[-c(1:4),]
metadata75<-metadata75[-c(22),]
metadata75<-metadata75[-c(48),]
metadata75<-metadata75[-c(21),]

metadata <- cbind(metadata75,metadata56)
metadata <- data.frame(t(metadata))

rowNames <- colnames(expression_df)
rownames(metadata)<-rowNames

# metadataExtra <-  read.table("data/GSE209964_series_matrix.txt", fill = TRUE, skip = 27, sep = "\t") #supplement dataset that contains, amongst others, 75 relevant samples
# 
# filter75 <- data.frame(colnames(expression_df[1:75]))
# toFilter <- data.frame(t(metadataExtra))
# colNames <- make.names(toFilter[1,],unique=TRUE)
# colnames(toFilter) <- colNames
# metadata75 <- filter(toFilter, X.Sample_geo_accession %in% filter75[,1])
# metadata75 <- data.frame(t(metadata75))
# metadata75 <- metadata75[-c(1:2),]
# 
# rm(filter75,metadataExtra,toFilter)
# 
# metadata<-cbind(metadata75, metadata56)
# metadata<- data.frame(t(metadata))
```


```{r}
ddset <- DESeqDataSetFromMatrix( 
    countData = expression_df, # Here we supply non-normalized count data 
    colData = metadata, # Supply the `colData` with our metadata data frame 
    design = ~ X.Sample_characteristics_ch1.4) # Supply our experimental variable to `design` 
vsd <- vst(ddset, blind=FALSE)

```
```{r}
plotPCA(vsd, intgroup=c( "X.Sample_characteristics_ch1.4"))
 
```

## 3
```{r}
set.seed(12345)

filtered_expression_df <- expression_df %>%
  dplyr::filter(rowSums(.) >= 1000)

deseq_object <- DESeq(ddset)
deseq_results <- results(deseq_object)

deseq_results <- lfcShrink(
  deseq_object, # The original DESeq2 object after running DESeq()
  coef = 2, # The log fold change coefficient used in DESeq(); the default is 2.
  res = deseq_results # The original DESeq2 results table
)

```
```{r}
deseq_df <- deseq_results %>%
  # make into data.frame
  as.data.frame() %>%
  # the gene names are row names -- let's make them a column for easy display
  tibble::rownames_to_column("Gene") %>%
  # add a column for significance threshold results
  dplyr::mutate(threshold = padj < 0.05) %>%
  
  dplyr::arrange(dplyr::desc(log2FoldChange))

 df<- (head( deseq_df))# table of differentially expressed genes
 df
 
 plotCounts(ddset, gene = "TSPAN6", intgroup = "X.Sample_characteristics_ch1.4")

readr::write_tsv(
  deseq_df,
  file.path(
    "data/results.tsv" 
  )
)


volcano_plot <- EnhancedVolcano::EnhancedVolcano(
  deseq_df,
  lab = deseq_df$Gene,
  x = "log2FoldChange",
  y = "padj",
  pCutoff = 0.01 # Loosen the cutoff since we supplied corrected p-values
)

volcano_plot


```


## 4
```{r}
library(devtools)
install_github("jokergoo/ComplexHeatmap")
library()

 df1<- expression_df %>% filter(rownames(expression_df) %in% (deseq_df[0:200,1]))
df1<- df1[order(rownames(df1)), ]

heatmap(data.matrix(df1), name ="Top 200 differentially expressed genes in long term survivors and non-long term survivors")
```


## 5
###i top go and gene ontology
```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.15")


BiocManager::install("topGO")
BiocManager::install("GO.db")
BiocManager::install("biomaRt")
BiocManager::install("Rgraphviz")
 
# Load the required R packages
library(topGO)
library(GO.db)
library(biomaRt)
library(Rgraphviz)
```
```{r}
filtered_expression_df
```



## 6

## 7

## 8

Note that the `echo = FALSE` parameter was added to the code chunk to
prevent printing of the R code that generated the plot.
